### ISSUE TITLE
default_value of InputObjectType isn't serialised like graphql-js breaking tools like Relay

### ISSUE THREAD

**Issue by mvgijssel** [rmosolgo/graphql-ruby#630]
When you currently define a field as InputObject with a default value other than an empty Hash the introspection query will return something like:
```json
"name": "order_by",
"description": null,
"type": {
  "kind": "INPUT_OBJECT",
  "name": "ResourceOrder",
  "ofType": null
},
"defaultValue": "{\"direction\":\"ASC\"}"
```
where the default value isn't an object but a JSON string (https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/introspection/input_value_type.rb#L21).

When running the introspection query on https://github.com/graphql/graphql-js it returns something slightly different (note the missing quotes around the key):
```json
"name": "order_by",
"description": null,
"type": {
  "kind": "INPUT_OBJECT",
  "name": "ResourceOrder",
  "ofType": null
},
"defaultValue": "{direction:\"ASC\"}"
```
This is because graphql-js isn't generating a JSON string but something else: https://github.com/graphql/graphql-js/blob/master/src/type/introspection.js#L344. This implementation calls the printer which prints without quotes around the keys: https://github.com/graphql/graphql-js/blob/master/src/language/printer.js#L83. 

Currently when using default_values in graphql-ruby we do the following to not break Relay:

```ruby
argument :order_by, -> { ResourceOrder }, default_value: NoJsonHash.new(direction: 'ASC')
```
where `NoJsonHash` implements `to_json` to always return {}. Which is a terrible fix 🙈

**Comment by mvgijssel** [comment]
After doing some more work realised that `NoJsonHash` totally doesn't work 😅. So the following "fix" works:
```ruby
module GraphQL
  module Language
    # @api private
    def self.serialize(value)
      if value.is_a? Hash
        JSON.generate({}, quirks_mode: true)
      else
        JSON.generate(value, quirks_mode: true)
      end
    end
  end
end
```
That way Relay doesn't crash but shouldn't be considered a solution.

**Comment by rmosolgo** [comment]
Derp, this one is always getting me! 

> defaultValue may return a String encoding (using the GraphQL language)

[spec](https://facebook.github.io/graphql/#sec-The-__InputValue-Type)

For scalars, this was fine, so I used `JSON`, but it broke enums (fixed recently #605) and apparently input objects are wrong too, figures! 

Thanks for reporting this, I wonder if a suitable fix is to modify that handle hashes specially, something like 

```ruby 
when Hash 
  "{#{value.map { |k, v| "#{k}: #{serialize(v)" }.join(",")}"
# ... 
```
